carservice/models.py

```diff
@@ -1,52 +1,223 @@
+        )
+
+
+def validate_year(prod_year):
+    first_made_car = date(1886, 1, 29)
+    if prod_year < first_made_car:
+        raise ValidationError(
+            _('You can\'t enter the year before the first car was made.'),
+            params={'prod_year': prod_year},
+        )
+    if prod_year > date.today():
+        raise ValidationError(
+            _('You can\'t add the car from future.'),
+            params={'prod_year': prod_year},
         )
 
 
+def check_car_exists(car_number):
+    if Car.objects.filter(serial_number=car_number).exists():
+        raise ValidationError(
+            _('Car with this VIN number already exists.'),
+            params={'car_number': car_number},
+        )
+
+
+def validate_mileage(mileage):
+    if mileage > 1000000:
+        raise ValidationError(
+            _('You can\'t add the car with more than 1 000 000 km.'),
+            params={'mileage': mileage},
+        )
+
+
+BRAND_CHOICES = (
+    ('Volkswagen', 'Volkswagen'),
+    ('BMW', 'BMW'),
+    ('Audi', 'Audi'),
+    ('Ford', 'Ford'),
+    ('Opel', 'Opel'),
+    ('Mercedes-Benz', 'Mercedes-Benz'),
+    ('Renault', 'Renault'),
+    ('Skoda', 'Skoda'),
+    ('Toyota', 'Toyota'),
+    ('Peugeot', 'Peugeot'),
+    ('Hyundai', 'Hyundai'),
+    ('Citroën', 'Citroën'),
+    ('Volvo', 'Volvo'),
+    ('Nissan', 'Nissan'),
+    ('Fiat', 'Fiat'),
+    ('Seat', 'Seat'),
+    ('Mazda', 'Mazda'),
+    ('Honda', 'Honda'),
+    ('Suzuki', 'Suzuki'),
+    ('Jeep', 'Jeep'),
+    ('Dacia', 'Dacia'),
+    ('Mitsubishi', 'Mitsubishi'),
+    ('MINI', 'MINI'),
+    ('Other', 'Other'),
+)
+
+
 class Car(models.Model):
-    serial_number = models.CharField(max_length=17, validators=[validate_ser_num])
-    car_mileage = models.PositiveIntegerField()
+    serial_number = models.CharField(max_length=17, validators=[check_vin_number, check_car_exists])
+    car_mileage = models.PositiveIntegerField(validators=[validate_mileage])
+    car_brand = models.CharField(max_length=30, choices=BRAND_CHOICES)
     car_model = models.CharField(max_length=30)
-    car_year = models.DateField()
+    date_of_prod = models.DateField(null=True, validators=[validate_year])
+
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+
 
     def __str__(self):
-        return f'Model: {self.car_model}, serial number:({self.serial_number})'
+        return f'Model: {self.car_model, self.car_brand}, serial number:({self.serial_number})'
+
+
+def car_available(vin_number):
+    if Offer.objects.filter(car=vin_number).exists():
+        raise ValidationError(
+            _('This car is already offered.'),
+            params={'vin_number': vin_number},
+        )
 
 
 class Offer(models.Model):
     description = models.TextField()
     price = models.FloatField(validators=[MinValueValidator(10.0)])
 
-    car = models.ForeignKey(Car, on_delete=models.CASCADE)
+    car = models.OneToOneField(Car, on_delete=models.CASCADE, validators=[car_available])
```

Comment:
skoro masz one to one to właśnie już nie trzeba sprawdzać, czy samochód jest dostępny, a przynajmniej nie na tej warstwie bo jeden samochód może mieć tylko jeden offer, ten komentarz tyczy się też innych miejsc w kodzie z tą relacją